pkg/server/ca/ca.go

```diff
@@ -0,0 +1,85 @@
+package ca
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"math/big"
+	"net/url"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/spiffe/spire/pkg/common/x509util"
+	"github.com/spiffe/spire/pkg/server/catalog"
+)
+
+type serverCAConfig struct {
+	Catalog     catalog.Catalog
+	TrustDomain url.URL
+	Backdate    time.Duration
+	DefaultTTL  time.Duration
+}
+
+type ServerCA interface {
+	SignX509SVID(ctx context.Context, csrDER []byte, ttl time.Duration) (*x509.Certificate, error)
+}
+
+type serverCA struct {
+	c      serverCAConfig
+	x509sn int64
+
+	mu sync.Mutex
```

Comment:
Should this be a pointer to a mutex? Feels a little safer IMO

Reply:
Also maybe better for RWMutex so that `SignX509SVID` can benefit?

Reply:
i tend to not do pointers to mutexes for structs that are constructed and consumed by pointer since it requires additional initialization and copying structs also requires judicious review of struct fields anyway to make sure aren't aliasing things you shouldn't (like maps and the like). If you feel strongly though....

I'm happy to switch to the RWMutex.